action.yml

```
@@ -117,7 +117,7 @@ runs:
 
         if [ -n "${{ inputs.extra-conf }}" ]; then
           if [ -n "${{ inputs.github-token }}" ]; then
-            export NIX_INSTALLER_EXTRA_CONF="${{ inputs.extra-conf }}access-tokens = github.com=${{ inputs.github-token }}"
+            export NIX_INSTALLER_EXTRA_CONF="${{ inputs.extra-conf }}"$'\n'"access-tokens = github.com=${{ inputs.github-token }}"
           else
             export NIX_INSTALLER_EXTRA_CONF="${{ inputs.extra-conf }}"
           fi
```

Despite suggesting `$'\n'`, I think I'd actually rather slightly refactor this to some less complicated bash (the nested if-else stuff makes my head hurt). This way it's easier to read, and harder to get wrong (see: our previous attempts at doing this... :sweat_smile: ):

```suggestion
        export NIX_INSTALLER_EXTRA_CONF_FILE=/tmp/extra-nix-config
        touch $NIX_INSTALLER_EXTRA_CONF_FILE

        if [ -n "${{ inputs.extra-conf }}" ]; then
          echo "${{ inputs.extra-conf }}" >> $NIX_INSTALLER_EXTRA_CONF_FILE
        fi

        if [ -n "${{ inputs.github-token }}" ]; then
          echo "access-tokens = github.com=${{ inputs.github-token }}" >> $NIX_INSTALLER_EXTRA_CONF_FILE
        fi
        
        export NIX_INSTALLER_EXTRA_CONF="$(cat $NIX_INSTALLER_EXTRA_CONF_FILE)"
```

(unfortunately, I can't select the entire range of these if-elses, sorry)

---

Agree!